crates/bevy_render/src/maths.wgsl

```diff
@@ -27,3 +27,28 @@ fn mat2x4_f32_to_mat3x3_unpack(
         vec3<f32>(a[1].zw, b),
     );
 }
+
+// Extracts the square portion of an affine matrix: i.e. discards the
+// translation.
+fn affine3_to_mat3x3(affine: mat4x3<f32>) -> mat3x3<f32> {
+    return mat3x3<f32>(affine[0].xyz, affine[1].xyz, affine[2].xyz);
+}
+
+// Returns the inverse of a 3x3 matrix.
+fn inverse_mat3x3(matrix: mat3x3<f32>) -> mat3x3<f32> {
+    let tmp0 = cross(matrix[1], matrix[2]);
+    let tmp1 = cross(matrix[2], matrix[0]);
+    let tmp2 = cross(matrix[0], matrix[1]);
+    let inv_det = 1.0 / dot(matrix[2], tmp2);
+    return transpose(mat3x3<f32>(tmp0 * inv_det, tmp1 * inv_det, tmp2 * inv_det));
+}
+
+// Returns the inverse of an affine matrix.
+//
+// Recall that an affine matrix is just a 4x4 matrix with the last column of [0,
+// 0, 0, 1]; thus the inverse is well-defined.
```

Comment:
I don't think the last column being 0,0,0,1 guarantees the matrix is invertible. For one thing the matrix determinant has to be non-zero, which means the determinant of the upper-left 3x3 matrix must be non-zero. Perhaps if the scale is 0,0,0, that would not be true?

Reply:
If the scale is zero, CPU gen would fail to generate the matrix too. I dont think scale will ever be zero, if one wants to hide a mesh thats what Visibility is for

Reply:
Right, I agree. I was only challenging the correctness of the "Recall that" part of the comment. I'm not sure that part of the comment is necessary really.

Reply:
I removed that part of the comment and replaced it with a Wikipedia link.